packages/grid/src/vaadin-grid.js

```diff
@@ -667,12 +638,74 @@ class Grid extends ElementMixin(
     }
 
     this.__intrinsicWidthCache = new Map();
+    // Cache the viewport rows to avoid unnecessary reflows while measuring the column widths
+    const fvi = this._firstVisibleIndex;
+    const lvi = this._lastVisibleIndex;
+    this.__viewportRowsCache = this._getVisibleRows().filter((row) => row.index >= fvi && row.index <= lvi);
```

Comment:
**nit:** I would expect that `_getVisibleRows()` already returns the visible rows. However, it instead seems to return the rendered rows, which isn't the same. The naming could be improved at some point.

```suggestion
    this.__viewportRowsCache = this._getVisibleRows().filter((row) => row.index >= fvi && row.index <= lvi);
```

Reply:
Yeah, the naming can be a bit misleading. But they are visible in the sense of not being `hidden` (even though they aren't necessarily inside the viewport).

Reply:
There's no limit to it. If you stretch your browser/grid vertically it renders more and more rows. Note that when you shrink the grid back, the added rows are not removed. Some of the rows may also be marked as `hidden` (for example if the item count goes below the number of row elements).